### PR TITLE
[Merged by Bors] - Re-added producer config to connector yaml

### LINKED ISSUE #2421: Expose producer options in connector config
Intent is to expose options that affect the performance of a connector

For example:
```yaml
version: 0.2.0
name: source-connector-draft
type: http
topic: cat-facts
direction: source
producer:
  linger: 1ms
  batch_size: 44
  compression: gzip
parameters:
  endpoint: https://catfact.ninja/fact
  interval: 10
  map: "catfact-map"
```

### PR DESCRIPTION
**Pull request by simlay**
This is the non breaking changes changes from #2426.

Closes #2421.

### CODE REVIEW

File: crates/fluvio-cli/src/connector/mod.rs [hunk 1]
```diff
@@ -92,40 +94,146 @@ pub struct ConnectorConfig {
...
     parameters: BTreeMap<String, String>,
+
     #[serde(default)]
     secrets: BTreeMap<String, SecretString>,
+
+    #[serde(default)]
+    producer: Option<ProducerParameters>,
+
+    #[serde(default)]
+    consumer: Option<ConsumerParameters>,
```
Comment (simlay) [inline crates/fluvio-cli/src/connector/mod.rs:111]: @morenol should I add a `smartmodule` section to this?

File: crates/fluvio-cli/src/connector/mod.rs [hunk 2]
```diff
@@ -92,40 +94,146 @@ pub struct ConnectorConfig {
...
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ConsumerParameters {
+    #[serde(default)]
+    partition: Option<i32>,
```
Comment (simlay) [inline crates/fluvio-cli/src/connector/mod.rs:117]: I'm tempted to add a `offset` field here but it's unclear how it should be specified. Thoughts?

Comment (sehz) [reply crates/fluvio-cli/src/connector/mod.rs:117]: should be the same as CLI but can be add later

File: crates/fluvio-cli/src/connector/mod.rs (hunk 1 above)
Comment (morenol) [inline crates/fluvio-cli/src/connector/mod.rs:111]: Do the feature flag for source and sink work as we would expect? @simlay

**Review by sehz — APPROVED** [review]
LGTM. We can add other changes later

File: crates/fluvio-cli/src/connector/mod.rs (hunk 1 above)
Comment (morenol) [reply crates/fluvio-cli/src/connector/mod.rs:111]: My bad, that is a fluvio-connectors thing

**Pull request closed without merging**